lib/hooks/useData.tsx

```diff
@@ -1,18 +1,46 @@
-import useSWR from 'swr';
-import * as fetch from '../utils/fetch';
+import { useMemo, useState } from 'react';
+import { ResponseWithStatus } from '../utils/fetch';
 
-type DataResponse<T> = {
-    data?: T;
-    isLoading: boolean;
-    isError?: Error;
-};
+type UseDataType<T> =
```

Comment:
The correct type for possibly failing requests is something that has separate `isSuccess` and `data` fields.